crates/biome_analyze/src/options.rs

```diff
@@ -55,34 +55,76 @@ impl AnalyzerRules {
 #[derive(Debug, Default)]
 pub struct AnalyzerConfiguration {
     /// A list of rules and their options
-    pub rules: AnalyzerRules,
+    rules: AnalyzerRules,
 
     /// A collections of bindings that the analyzers should consider as "external".
     ///
     /// For example, lint rules should ignore them.
-    pub globals: Vec<Box<str>>,
+    globals: Vec<Box<str>>,
 
     /// Allows to choose a different quote when applying fixes inside the lint rules
-    pub preferred_quote: PreferredQuote,
+    preferred_quote: PreferredQuote,
 
     /// Indicates the type of runtime or transformation used for interpreting JSX.
-    pub jsx_runtime: Option<JsxRuntime>,
+    jsx_runtime: Option<JsxRuntime>,
+}
+
+impl AnalyzerConfiguration {
+    pub fn with_rules(mut self, rules: AnalyzerRules) -> Self {
+        self.rules = rules;
+        self
+    }
+
+    pub fn with_globals(mut self, globals: Vec<Box<str>>) -> Self {
+        self.globals = globals;
+        self
+    }
+
+    pub fn with_jsx_runtime(mut self, jsx_runtime: JsxRuntime) -> Self {
+        self.jsx_runtime = Some(jsx_runtime);
+        self
+    }
+
+    pub fn with_preferred_quote(mut self, preferred_quote: PreferredQuote) -> Self {
+        self.preferred_quote = preferred_quote;
+        self
+    }
+
+    /// Returns the rules that are configured for this analyzer
+    pub fn rules(&self) -> &AnalyzerRules {
+        &self.rules
+    }
```

Comment:
```suggestion
    /// Returns the rules that are configured for this analyzer
    pub fn rules(&self) -> &AnalyzerRules {
        &self.rules
    }
```

I think it's both simpler and more flexible to let the `rules` field remain public and then you wouldn't need this getter.

Rust ownership rules already prevent you from messing with instances you may not mutate, and if there are no additional business rules to enforce (there don't seem to be any here), then there's no real reason to do it like this.